### PR TITLE
add "GeneratedBy"s in `dataset_description.json` for BIDS 1.4+ compliance

### LINKED ISSUE #1018: BIDS v1.4+ requires "GeneratedBy" instead of "PipelineDescription" in `dataset_description.json`
Similar to https://github.com/PennLINC/qsiprep/issues/630.

Current behavior generate a `UserWarning` when instantiating a `GroupAFQ` object:
```
...bids/layout/validation.py:122: UserWarning: The PipelineDescription field was superseded by GeneratedBy in BIDS 1.4.0. You can use ``pybids upgrade`` to update your derivative dataset.
  warnings.warn("The PipelineDescription field was superseded "
```

### PR DESCRIPTION
**Pull request by pierre-nedelec**
Fixes #1018

### CODE REVIEW

File: AFQ/api/group.py [hunk 1]
```diff
@@ -187,7 +187,9 @@ def __init__(self,
             "Name": bids_description["Name"],
             "BIDSVersion": bids_description["BIDSVersion"],
             "PipelineDescription": {"Name": "pyAFQ",
-                                    "Version": pyafq_version}}
+                                    "Version": pyafq_version},
+            "GeneratedBy": [{"Name": op.basename(self.afq_path),
+                             "Version": pyafq_version}]}
```
Comment (arokem) [inline AFQ/api/group.py:192]: Can you really have both of these side-by-side?

**Review by arokem — COMMENTED** [review]
We had this in a previous PR, that was eventually closed: https://github.com/yeatmanlab/pyAFQ/pull/870. I think that we were waiting for https://github.com/bids-standard/pybids/issues/864 to get merged in pybids first, but if this now works without that, I would be OK making this change now.

File: AFQ/api/group.py (hunk 1 above)
Comment (arokem) [inline AFQ/api/group.py:192]: Also, while we're here, could we add something like: 
```suggestion
                                    "Version": pyafq_version},
            "GeneratedBy": [{"Name": op.basename(self.afq_path),
                             "Version": pyafq_version}], 
            "HowToAcknowledge": "Please cite this paper: https://apertureneuro.org/article/77465-evaluating-the-reliability-of-human-brain-white-matter-tractometry"}
```

**Review by arokem — APPROVED** [review]
Looks like this is passing the CI tests, so I will go ahead and merge it.

File: AFQ/api/group.py (hunk 1 above)
Comment (pierre-nedelec) [reply AFQ/api/group.py:192]: I left the PipelineDescription for compatibility. I think it's just going to be extra and should be deleted in the near future.

**Pull request merged**